k2/python/csrc/torch/index_select.cu

```diff
@@ -0,0 +1,221 @@
+/**
+ * @brief Index select for k2.
+ *
+ * Unlike torch.index_select, when an entry is -1, it sets
+ * the destination entry to 0.
+ *
+ * @copyright
+ * Copyright (c)  2020  Mobvoi Inc.        (authors: Fangjun Kuang)
+ *
+ * @copyright
+ * See LICENSE for clarification regarding multiple authors
+ */
+
+#include "k2/csrc/context.h"
+#include "k2/python/csrc/torch/index_select.h"
+#include "k2/python/csrc/torch/torch_util.h"
+#include "torch/extension.h"
+
+namespace k2 {
+
+template <typename T>
+static torch::Tensor IndexSelect1D(torch::Tensor src, torch::Tensor index) {
```

Comment:
The functions in this file still need documentation!  Including the wrappers!   Does n't have to be very long/detailed b ut should exist.

Reply:
Fixed.